OpenRA.Game/Graphics/WorldRenderer.cs

```diff
@@ -133,7 +133,9 @@ void GenerateRenderables()
 			foreach (var e in World.ScreenMap.RenderableEffectsInBox(Viewport.TopLeft, Viewport.BottomRight))
 				renderablesBuffer.AddRange(e.Render(this));
 
-			foreach (var renderable in renderablesBuffer.OrderBy(RenderableScreenZPositionComparisonKey))
+			renderablesBuffer.Sort((x, y) => RenderableZPositionComparisonKey(x).CompareTo(RenderableZPositionComparisonKey(y)));
```

Comment:
A `// PERF: Avoid LINQ` here might also be wise to prevent someone from turning this back in the future.

Reply:
The whole method is already tagged with such a comment.

Reply:
So it is.